google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+from google.api_core import exceptions
+from google.cloud.aiplatform.metadata.metadata_store import _MetadataStore
+from google.cloud.aiplatform.metadata.context import _Context
+from google.cloud.aiplatform.metadata.execution import _Execution
+from google.cloud.aiplatform.metadata.artifact import _Artifact
+
+
+class _MetadataService:
+    """Contains the exposed APIs to interact with the Managed Metadata Service."""
+
+    def __init__(self):
+        self._experiment = None
+        self._run = None
+        self._metrics = None
+
+    # TODO(nachocano) Error handling this way is not very readable. Find a better way.
+    def set_experiment(self, experiment_name):
+        if not experiment_name:
+            raise ValueError(f"Invalid experiment_name {experiment_name}.")
+        try:
+            _MetadataStore()
+        except exceptions.NotFound:
+            logging.info(
+                f"Creating a default MetadataStore for experiment {experiment_name}"
+            )
+            _MetadataStore.create()
+
+        try:
+            context = _Context(context_name=experiment_name)
+        except exceptions.NotFound:
```

Comment:
@sasha-gitg I cannot catch this exception. I'm getting something like:

```---------------------------------------------------------------------------
_InactiveRpcError                         Traceback (most recent call last)
~/Library/Python/3.6/lib/python/site-packages/google/api_core/grpc_helpers.py in error_remapped_callable(*args, **kwargs)
     72         try:
---> 73             return callable_(*args, **kwargs)
     74         except grpc.RpcError as exc:

~/Library/Python/3.6/lib/python/site-packages/grpc/_channel.py in __call__(self, request, timeout, metadata, credentials, wait_for_ready, compression)
    922                                       wait_for_ready, compression)
--> 923         return _end_unary_response_blocking(state, call, False, None)
    924 

~/Library/Python/3.6/lib/python/site-packages/grpc/_channel.py in _end_unary_response_blocking(state, call, with_call, deadline)
    825     else:
--> 826         raise _InactiveRpcError(state)
    827 

_InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.NOT_FOUND
	details = "Resource not found.; GetContext is unable to read context resource from auxiliary table with name: projects/354090969940/locations/us-central1/metadataStores/default/contexts/experiment"
	debug_error_string = "{"created":"@1617954685.100137000","description":"Error received from peer ipv6:[2607:f8b0:400a:807::200a]:443","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Resource not found.; GetContext is unable to read context resource from auxiliary table with name: projects/354090969940/locations/us-central1/metadataStores/default/contexts/experiment","grpc_status":5}"
```

I believe this is being thrown from `getattr(self.api_client, self._getter_method)` in resources.py.
I used to have something similar before for the create call, and then end up calling `.result()` and it worked. Is it because it is returning a future? How can I make it sync?

Reply:
Ohh ok, indeed, calling `.result()` works...

Reply:
hmm, not really, still getting this exception

Reply:
OK, figured it out... resolving...